tc/core/cuda/cuda_rtc.cc

```diff
@@ -60,30 +64,87 @@ void checkOrCreateContext() {
   }
 }
 
-std::unique_ptr<CudaRTCFunction> CudaRTCFunction::Compile(
-    const std::string& name,
-    const std::string& source) {
-  std::unique_ptr<CudaRTCFunction> res(new CudaRTCFunction());
-  res->specializedName = name;
-  res->cleared_ = false;
-
-  if (FLAGS_debug_tc_mapper) {
-    LOG(INFO) << "NVRTC function source:\n" << source;
-  }
-  // Actually do the compiling.
-  nvrtcProgram prog;
-  TC_NVRTC_CHECK(
-      nvrtcCreateProgram(&prog, source.c_str(), nullptr, 0, nullptr, nullptr));
-
-  // Get the architecture of the current device.
-  int device, minor, major;
+namespace {
+static std::tuple<int, int, int> getCudaArchitecture() {
+  int device, major, minor;
   CUdevice deviceHandle;
   TC_CUDA_RUNTIMEAPI_ENFORCE(cudaGetDevice(&device));
   TC_CUDA_DRIVERAPI_ENFORCE(cuDeviceGet(&deviceHandle, device));
   TC_CUDA_DRIVERAPI_ENFORCE(cuDeviceGetAttribute(
       &major, CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR, deviceHandle));
   TC_CUDA_DRIVERAPI_ENFORCE(cuDeviceGetAttribute(
       &minor, CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR, deviceHandle));
+  return std::tuple<int, int, int>(device, major, minor);
+}
+
+static std::string llvmCompile(
+    const std::string& name,
+    const std::string& source) {
+  int device, major, minor;
+  std::tie(device, major, minor) = getCudaArchitecture();
+
+  std::string pat("/tmp/cudaXXXXXX");
+  std::vector<char> ifn(pat.begin(), pat.end());
+  TC_CHECK_GE(mkstemp(ifn.data()), 0); // string.c_str is const char*
+  std::string inputFileName(ifn.begin(), ifn.end());
+  tc::ScopeGuard sgi([&]() { std::remove(inputFileName.c_str()); });
```

Comment:
Why do you need all these genuflections with remove?

Reply:
Because, as indicated, string.c_str is const char*.
I could const cast to save 2 lines of code, lmk if you prefer.

Reply:
This is `<cstdio>`'s `std::remove(const char*)`, not the one coming from `<algorithm>`.
It deletes the underlying file, I'll add a comment.